src/utils/Accessibility.js

```diff
@@ -0,0 +1,19 @@
+/*
+Copyright 2019 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+export function focusCapturedRef(ref) {
+    if (ref) ref.focus();
```

Comment:
Huh, so this is focusing... the list on each render? Please add a comment about where and when you'd expect to use this and why (as I don't really get it yet).

Reply:
Refs are captured at most twice: the first call can be null for some reason, and the second is an element. Further renders don't cause this to be called.

Will add comment.

Reply:
Hmm, but why is the first render with an element the right time to call `focus`? It feels a bit like we're mixing the intent you want to achieve with details of how React calls ref callback, which I believe they'd be free to change in future versions.

Reply:
It was called a hack for a reason. There's no reliable way to detect first render afaik

Reply:
I think I am mostly still missing _why_ we want to do the `focus` at all... Once I understand the reason, my comments should get more useful. 😅

Reply:
It's mentioned briefly in https://github.com/matrix-org/matrix-react-sdk/commit/2a187810fd66d7d45bcccf0f8fb64691fd0578c6 - it's so people can use they keyboard to navigate the list. Without the focus, your cursor is left wherever it last was (composer, random part of the page, etc)

Reply:
Ah okay... I did read the commit message but for some reason it didn't click in my head until now... 😅 Could you use a `componentDidMount` hook instead? I feel like that would make the intent more obvious here.

Reply:
The ref is null at that point :(

Reply:
Hmm... The reason it feels strange to me (`componentDidMount` would also have this issue if it had worked) is that this only happens to work at all because of a separate component `TopLeftMenuButton` which doesn't render this component when the menu is collapsed. If `TopLeftMenuButton` were changed in the future to always render but hide the menu with CSS, then this `focus` call is going to happen at some random time on page load, and I would find that quite unexpected.

What if the `focus` call moved over to `TopLeftMenuButton`'s `onToggleMenu`? That would make it clear that we want to focus when the menu toggles, not at whatever magical time React happens to mount the elements.

Reply:
We'd still have to use the ref hack so we could detect the mounting, I think. Worth a shot though.